core/base/src/TDirectory.cxx

```diff
@@ -1243,27 +1245,26 @@ Int_t TDirectory::SaveObjectAs(const TObject *obj, const char *filename, Option_
 {
    if (!obj) return 0;
    Int_t nbytes = 0;
-   TString fname = filename;
-   if (!filename || !filename[0]) {
-      fname.Form("%s.root",obj->GetName());
-   }
-   TString cmd;
+   TString fname, opt = option, cmd;
+   if (filename && *filename)
+      fname = filename;
+   else
+      fname.Form("%s.root", obj->GetName());
+   opt.ToLower();
+
    if (fname.Index(".json") > 0) {
-      cmd.Form("TBufferJSON::ExportToFile(\"%s\",(TObject*) %s, \"%s\");", fname.Data(), TString::LLtoa((Longptr_t)obj, 10).Data(), (option ? option : ""));
+      cmd.Form("TBufferJSON::ExportToFile(\"%s\", (TObject *) 0x%zx, \"%s\");", fname.Data(), (size_t) obj, (option ? option : ""));
       nbytes = gROOT->ProcessLine(cmd);
    } else {
-      cmd.Form("TFile::Open(\"%s\",\"recreate\");",fname.Data());
+      cmd.Form("TFile::Open(\"%s\",\"%s\");", fname.Data(), opt.Contains("a") ? "update" : "recreate");
```

Comment:
Why are we going through `ProcessLine` rather than calling those directly? (asking even-though it is pre-existing).

Reply:
It was easy "plugin" mechanism. To avoid dependency from `RIO` library here, in core